src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryCriteriaLocationsValidatorTests.cs

```diff
@@ -0,0 +1,227 @@
+using FluentValidation.TestHelper;
+using GovUk.Education.ExploreEducationStatistics.Common.Validators;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Requests;
+
+public class DataSetQueryCriteriaLocationsValidatorTests
+{
+    private readonly DataSetQueryCriteriaLocations.Validator _validator = new();
+
+    public static readonly TheoryData<DataSetQueryLocation> ValidLocationsSingle = new()
+    {
+        new DataSetQueryLocationId { Level = "NAT", Id = "12345" },
+        new DataSetQueryLocationCode { Level = "NAT", Code = "E92000001" },
+        new DataSetQueryLocationId { Level = "REG", Id = "12345" },
+        new DataSetQueryLocationCode { Level = "REG", Code = "E12000003" },
+        new DataSetQueryLocationId { Level = "LA", Id = "12345" },
+        new DataSetQueryLocationLocalAuthorityCode { Code = "E08000019" },
+        new DataSetQueryLocationLocalAuthorityCode { Code = "E09000021 / E09000027" },
+        new DataSetQueryLocationLocalAuthorityOldCode { OldCode = "373" },
+        new DataSetQueryLocationLocalAuthorityOldCode { OldCode = "314 / 318" },
+        new DataSetQueryLocationId { Level = "SCH", Id = "12345" },
+        new DataSetQueryLocationSchoolUrn { Urn = "107029" },
+        new DataSetQueryLocationSchoolLaEstab { LaEstab = "3732060" },
+        new DataSetQueryLocationId { Level = "PROV", Id = "12345" },
+        new DataSetQueryLocationProviderUkprn { Ukprn = "10066874" },
```

Comment:
Nitpick but might be nice to organise these so that the same types are next to each other. Makes it easier to digest which scenarios are covered